Update ChatServer.java

dimits-ts opened the pull request.

Commit:
Update ChatServer.java

Commit:
Update Logging.java

Commit:
Update Server.java

Commit:
Update GameServer.java

Commit:
Update GameEngine.java

AAAlex-123 commented on Logging.java:
```diff
@@ -13,17 +17,29 @@
 	 * @param text String, text to send
 	 * @param args Object[], arguments
 	 */
-	default void log(String text, Object... args) {
-		System.out.printf(text + "\n", args);
+	default void log(String text) {
+		System.out.println(text);
 	}
```
γτ να το αλλάξεις; υπήρχε κάπου πρόβλημα;

dimits-ts replied:
δες κατω

AAAlex-123 commented on Server.java:
```diff
@@ -196,17 +200,41 @@ protected void broadcast(String msg, Object... args) {
+				logerr("Error in broadcast()\n"+ (printStackTrace ?e.toString():""));
 			} catch (NullPointerException e) {
-				;
+				if(printStackTrace) logerr(e.toString());
```
δε μας ενδιαφέρει αυτό το λάθος, γι' αυτό δεν είχα βάλει logerr. κάθε φορά που κάνει broadcast θα συμβαίνει αυτό το exception είναι αναμενόμενο

dimits-ts replied:
ok

AAAlex-123 commented on Server.java:
```diff
@@ -196,17 +200,41 @@ protected void broadcast(String msg, Object... args) {
-	
-
+	@SuppressWarnings("serial")
+	protected class Screen extends JFrame{
```
γιατί έχει ο server ένα screen

βάζε σχόλια ριιιιιιιιιιιιιιιιιιιιιιιιιιιιιιιι

dimits-ts replied:
για να ξερει ο χρηστης τι γινεται

AAAlex-123 replied:
α αυτό με τα logs οκ

AAAlex-123 commented on Logging.java:
```diff
@@ -13,17 +17,29 @@
 	 */
-	default void logerr(String text, Object... args) {
-		System.err.printf(text + "\n", args);
+	default void logerr(String error_msg) {
```
γτ να μην είναι logerr(String error_msg, Object... args);

dimits-ts replied:
γινονταν υπερβολικα σπαγγετι σε καποια σημεια, η logerr θα επρεπε να ειχε 2 συνεχομενα printfs για να τα πιανει ολα. Ετσι ειναι πολυ πιο ευκολο να γινει η αντικατασταση(τουλαχιστον για εμενα που επρεπε να το αλλαξω σε ολα τα αρχεια). Προτιμω να γραφω 2 λεξεις και να ξερω 100% τι θα παρει η μεθοδος

AAAlex-123 replied:
εχ οκ σουρ, αν και το μόνο που αλλάζει όταν την καλείς στον κώδικα είναι το String.format στην αρχή, το υπόλοιπο είναι ακριβώς το ίδιο

AAAlex-123 commented on GameServer.java:
```diff
@@ -85,18 +77,13 @@ public void run() {
-				;
-			}
+			logerr("BindException while setting up server; a server is already running on this port\n"+ (printStackTrace ?e.toString():""));
+			System.exit(1);
```
αν το κάνεις έτσι το παράθυρο cmd θα κλείσει μόλις εμφανιστεί το μήνυμα λάθους από το logerr κ ο χρήστης δε θα προλάβει να το διαβάσει άρα δε θα ξέρει πού είναι το πρόβλημα, γι' αυτό είχα βάλει το while (true)

dimits-ts replied:
Δεν το ειχα σκεφτει ετσι οπως το λες. Μπορουμε να βαλουμε JOptionPane.ERROR τοτε, ωστε να κανει exit μολις πατησει το Χ ο χρηστης.

AAAlex-123 replied:
χμ i guess για έκτακτες ειδοποιήσεις όπως το ότι δεν μπορεί να τρέξει μπορούμε να βάλουμε popuρ αντί για ένα ακόμη log, σουρ

AAAlex-123 commented on Server.java:
```diff
@@ -183,7 +182,12 @@ public void actionPerformed(ActionEvent e) {
 	}
-
+	
+	@Override 
+	public void log(String text) {
+		screen.pushMessage(text);
+	}
+	
```
γτ το log πάει σε screen κ το logerr σε αρχείο; κανένα από τα δύο δεν αφορά τον χρήστη, και τα δύο είναι logs. απλά το ένα είναι για να βλέπουμε την πορεία του προγράμματος (πχ άρχισε γύρος, ήρθε ένα τσατ, ο Χ έπαιξε) κ το άλλο είναι για τα exceptions.
Αν τα θέλουμε για εμάς, για debugging όταν θα τρέχει με exe, μπορούμε να βάλουμε μια boolean αν θα πηγαίνουν όλα τα log σε screen ή σε αρχείο (screen για δικό μας debug, αρχείο για τον χρήστη)

dimits-ts replied:
το logerr ειναι το standard για ολα τα αρχεια, αν θελεις τα λαθη να πηγαινουν στην οθονη μπορουμε να την κανουμε override.

AAAlex-123 replied:
ναι αλλά γιατί είναι διαφορετικό από το log; και τα δύο κάνουν log. logerr για σφάλματα, log για τα υπόλοιπα

AAAlex-123 reviewed (commented):
διάβασε τα σχόλια :ρ

Commit:
Update Logging.java

updated loggerr

kill me

Commit:
Update Logging.java

AAAlex-123 committed:
playerLabel now updates

added `public updateXXXXConnectionCounter(int)` to update the playerLabel JLabel when a player joins/leaves the chat/game servers

AAAlex-123 committed:
added methods to update label

see Server commit

AAAlex-123 committed:
server now changes screen, logerr fix

Commit:
Update ChatServer.java

Commit:
Update Logging.java

Commit:
Update GameEngine.java

Commit:
Update GameServer.java

AAAlex-123 reviewed (approved):
`n i c e`

AAAlex-123 reviewed (approved):
`nice`
